examples/cosmos/ics20/denomTrace.qnt

```diff
@@ -173,6 +188,87 @@ module denomTraceTest {
   }))
 }
 
+module denomTraceSimulation {
+  import denomTrace.*
+
+  pure val chains = Set("A", "B", "C")
+  pure val transferSequence = List(
+    ("A", "B"),
+    ("B", "C"),
+    ("C", "A"),
+    ("A", "C"),
+    ("C", "B"),
+    ("B", "A")
+  )
+
+  const channelPredicate: (str -> str -> str) => bool
+
+  pure val channelNames = Set("channel-1", "channel-2", "channel-3")
+
+  /// All possible combination of channels by chain satisfying `channelPredicate`
+  pure val possibleChannelSetups = chains.setOfMaps(chains.setOfMaps(channelNames)).filter(channelPredicate)
+
+  var trace: DenomTrace
+  var channelSetup: str -> str -> str
+  var counter: int
+
+  def getChannelsForTransfer(transfer: (str, str)): { sourceChannel: str, destChannel: str } =
+    ({
+      sourceChannel: channelSetup.get(transfer._1).get(transfer._2),
+      // The destination channel would be the source channel in the opposite
+      // direction, so we only need to reverse the order of the chains
+      destChannel: channelSetup.get(transfer._2).get(transfer._1),
+    })
+
+  action init =
+    nondet someMap = possibleChannelSetups.oneOf()
+    all {
+      trace' = {
+        path: [],
+        baseDenom: "denom",
+      },
+      channelSetup' = someMap,
+      counter' = 0,
+    }
+
+  action step =
+    val channels = getChannelsForTransfer(transferSequence[counter])
+    all {
+      trace' = updateTrace(trace, "transfer", channels.sourceChannel, "transfer", channels.destChannel),
+      counter' = counter + 1,
+      channelSetup' = channelSetup,
+    }
+
+  action unchanged = all {
+    trace' = trace,
+    counter' = counter,
+    channelSetup' = channelSetup,
+  }
+
+  run simulationTest = init.then(step.repeated(6)).then((all {
+    def show(a) = true { show(channelSetup) },
+    assert(trace.path == []),
+    unchanged,
+  }))
+}
+
+// Fails because the channel setup is not proper
+// The aim is to show that it is, in fact, necessary to enforce `isProperChannelSetup`
+module randomChannelsPBT {
```

Comment:
```suggestion
module randomChannelsTests {
```

Not everyone knows what PBT means, and it's also less important that it's about tests